frontend/src/scenes/session-recordings/player/inspector/components/ItemPerformanceEvent.tsx

```diff
@@ -136,6 +136,33 @@ function renderTimeBenchmark(milliseconds: number): JSX.Element {
     )
 }
 
+function itemSizeInfo(item: PerformanceEvent): {
+    bytes: string
+    compressionPercentage: number | null
+    decodedBodySize: string | null
+    encodedBodySize: string | null
+    formattedCompressionPercentage: string | null
+    isFromLocalCache: boolean
+} {
+    const bytes = humanizeBytes(item.encoded_body_size || item.decoded_body_size || item.transfer_size || 0)
+    const decodedBodySize = item.decoded_body_size ? humanizeBytes(item.decoded_body_size) : null
+    const encodedBodySize = item.encoded_body_size ? humanizeBytes(item.encoded_body_size) : null
+    const compressionPercentage =
+        item.decoded_body_size && item.encoded_body_size
+            ? ((item.decoded_body_size - item.encoded_body_size) / item.decoded_body_size) * 100
+            : null
+    const formattedCompressionPercentage = compressionPercentage ? `${compressionPercentage.toFixed(1)}%` : null
+    const isFromLocalCache = item.transfer_size === 0 && (item.decoded_body_size || 0) > 0
```

Comment:
I'm unsure if this would work for all platforms, some libs/FWs don't allow reading the bodies twice, so you can't do it using an interceptor otherwise the app final consumer won't be able to read it as well.
In this case, the SDK might fall back to 0 just because it's not possible and/or unknown (maybe the solution here is to just omit it), but the FE will consider it `isFromLocalCache=true` which h is wrong.

Maybe a better option is to introduce a new flag called `isFromLocalCache=true`.

Not a blocker, just to consider, maybe adding a comment is also fine, so people are aware to omit the sizes instead of zero.

Reply:
Yeah, JS doesn't let you read bodies twice but does let you clone them - feels like a distinction you could hide from humans 🤣 🤷 

We're lucky that browsers dump this info out as standard for us to scoop up